clients/venice-pulsar/src/main/java/com/linkedin/venice/pulsar/sink/VeniceSink.java

```diff
@@ -0,0 +1,248 @@
+package com.linkedin.venice.pulsar.sink;
+
+import static com.linkedin.venice.CommonConfigKeys.SSL_ENABLED;
+import static com.linkedin.venice.samza.VeniceSystemFactory.DEPLOYMENT_ID;
+import static com.linkedin.venice.samza.VeniceSystemFactory.DOT;
+import static com.linkedin.venice.samza.VeniceSystemFactory.SYSTEMS_PREFIX;
+import static com.linkedin.venice.samza.VeniceSystemFactory.VENICE_AGGREGATE;
+import static com.linkedin.venice.samza.VeniceSystemFactory.VENICE_CONTROLLER_DISCOVERY_URL;
+import static com.linkedin.venice.samza.VeniceSystemFactory.VENICE_PUSH_TYPE;
+import static com.linkedin.venice.samza.VeniceSystemFactory.VENICE_ROUTER_URL;
+import static com.linkedin.venice.samza.VeniceSystemFactory.VENICE_STORE;
+
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.samza.VeniceSystemFactory;
+import com.linkedin.venice.samza.VeniceSystemProducer;
+import com.linkedin.venice.utils.Utils;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.pulsar.client.api.schema.GenericObject;
+import org.apache.pulsar.common.schema.KeyValue;
+import org.apache.pulsar.functions.api.Record;
+import org.apache.pulsar.io.core.Sink;
+import org.apache.pulsar.io.core.SinkContext;
+import org.apache.samza.config.MapConfig;
+
+
+/**
+ * A Pulsar Sink that sends messages to Venice.
+ *
+ * Please refer to Apache Pulsar's documentation for more information on Pulsar connectors:
+ * https://pulsar.apache.org/docs/2.10.x/io-use/
+ *
+ * Available configuration parameters: see VeniceSinkConfig class.
+ */
+public class VeniceSink implements Sink<GenericObject> {
```

Comment:
VeniceSink goes in line with Pulsar's naming of sink components ("Pulsar" is redundant in that case)

Reply:
in the Pulsar IO ecosystem you usually name this component like: CassandraSink, ElastichSearchSink, MongoDBSink, JDBCSink.
So this name matches the conventions

Reply:
We should keep this Sink here as it is really next to the Samza client.

In the case of Samza the SystemFactory is named `VeniceSystemFactory`, so it is consistent to have the Sink named `VeniceSink`

I guess that someday we will have a Flink "VeniceSink"

I suggest to follow the conventions of each project Samza, Pulsar, Flink...


That said, from the Pulsar user's point of view the classname is not important as you give a logical name to the sink, in this case it will be 'venice'
`pulsar-admin sinks create -type venice .....`

Reply:
I guess the thing is that "sink" is a fairly generic and widely used term. As you pointed out Enrico, Flink also has sinks.

If there was a `VeniceSink` inside a Pulsar repo, and a different `VeniceSink` in some Flink repo, then it would be obvious from the enclosing context that the former is a Pulsar sink and the latter a Flink sink. But if all these sinks are in the Venice repo, it might be a little confusing (not all users necessarily pay attention to the FQCN).

Regarding the `VeniceSystemFactory`, it is a good point that Samza is missing from there. I guess it didn't occur to us at the time given that we had only one stream processor to deal with back then. I would not mind adding Samza to that class name to make things standardized and unambiguous. IOW, perhaps the ideal end state is something along the lines of:

- `VeniceSamzaSystemFactory`
- `VenicePulsarSink`
- `VeniceFlinkSink`
- and so on

Would that be acceptable?

Reply:
+1 for VenicePulsarSink

Reply:
Fixed